app/src/main/java/com/mapbox/maps/testapp/examples/snapshotter/MapViewSnapshotActivity.kt

```diff
@@ -25,8 +25,11 @@ class MapViewSnapshotActivity : AppCompatActivity() {
     mapboxMap.loadStyleUri(Style.MAPBOX_STREETS)
 
     binding.fab.setOnClickListener {
-      val bitmap = binding.mapView.snapshot()
-      binding.imageView.setImageBitmap(bitmap)
+      binding.mapView.snapshot { bitmap ->
```

Comment:
Used async version of snapshotter to avoid blocking main thread

Reply:
very solid solution!